crates/driver/src/infra/blockchain/gas.rs

```diff
@@ -32,27 +33,32 @@ impl GasPriceEstimator {
         let additional_tip = mempools
             .iter()
             .find(|mempool| matches!(mempool.kind, mempool::Kind::MEVBlocker { .. }))
-            .map(|mempool| {
-                (
-                    match mempool.kind {
-                        mempool::Kind::MEVBlocker {
-                            max_additional_tip, ..
-                        } => max_additional_tip,
-                        _ => unreachable!(),
-                    },
-                    mempool.additional_tip_percentage,
-                )
+            .map(|mempool| match mempool.kind {
+                mempool::Kind::MEVBlocker {
+                    max_additional_tip,
+                    additional_tip_percentage,
+                    ..
+                } => (max_additional_tip, additional_tip_percentage),
+                _ => unreachable!(),
```

Comment:
This `find if matches!()` and then `map` using `unreachable!()` seems a bit weird (already before this PR).
Alternatively you could consider:
```rust
.filter_map(|mempool| match mempool.kind {
    MEVBlocker => Some((max_tip, additional_tip)),
    _ => None
})
.next();
```